AlfrescoKit/AlfrescoKit/Picking View Controller/AKAlfrescoNodePickingListViewController.m

```diff
@@ -0,0 +1,258 @@
+}
+
+- (instancetype)initAlfrescoFolderPickerWithRootFolder:(AlfrescoFolder *)folder
+                                         selectedNodes:(NSMutableArray *)selectedNodes
+                                              delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                        listingContext:(AlfrescoListingContext *)listingContext
+                                               session:(id<AlfrescoSession>)session
+{
+    self = [self initWithAlfrescoFolder:folder listingContext:listingContext session:session];
+    if (self)
+    {
+        self.nodePickerOptions = AKAlfrescoNodePickerTypeFolderPicker;
+        self.delegate = delegate;
+        self.selectedNodes = (selectedNodes) ? selectedNodes : [NSMutableArray array];
+    }
+    return self;
+}
+
+- (instancetype)initAlfrescoDocumentPickerWithRootFolder:(AlfrescoFolder *)folder
+                                       multipleSelection:(BOOL)allowMultiple
+                                           selectedNodes:(NSMutableArray *)selectedNodes
+                                                delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                                 session:(id<AlfrescoSession>)session
+{
+    return [self initAlfrescoDocumentPickerWithRootFolder:folder multipleSelection:allowMultiple selectedNodes:selectedNodes delegate:delegate listingContext:nil session:session];
+}
+
+- (instancetype)initAlfrescoDocumentPickerWithRootFolder:(AlfrescoFolder *)folder
+                                       multipleSelection:(BOOL)allowMultiple
+                                           selectedNodes:(NSMutableArray *)selectedNodes
+                                                delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                          listingContext:(AlfrescoListingContext *)listingContext
+                                                 session:(id<AlfrescoSession>)session
+{
+    self = [self initWithAlfrescoFolder:folder listingContext:listingContext session:session];
+    if (self)
+    {
+        self.nodePickerOptions = AKAlfrescoNodePickerTypeDocumentPicker;
+        if (allowMultiple)
+        {
+            self.nodePickerOptions |= AKAlfrescoNodePickerTypeMultiplePicker;
+        }
+        self.delegate = delegate;
+        self.selectedNodes = (selectedNodes) ? selectedNodes : [NSMutableArray array];
+        
+    }
+    return self;
+}
+
+- (NSString *)nibName
+{
+    return NSStringFromClass([self superclass]);
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    // Setup the toolbar
+    [self setupToolbar];
+    // Display the toolbar if required
+    BOOL isFolderPicker = self.nodePickerOptions & AKAlfrescoNodePickerTypeFolderPicker;
+    BOOL isMultipleDocumentPicker = self.nodePickerOptions & AKAlfrescoNodePickerTypeDocumentPicker && self.nodePickerOptions & AKAlfrescoNodePickerTypeMultiplePicker;
```

Comment:
Might be worth adding parentheses to this, or splitting it across two lines - it's easy to miss the bitmasking here (or was for me at least!)